### PR TITLE
Modifiers for PDT II, "Subtle Blow" II and "Cure" potency II

### PR DESCRIPTION
**Pull request by Fizztastic**

### CODE REVIEW

File: src/map/utils/battleutils.cpp [hunk 1]
```diff
@@ -2020,7 +2020,7 @@ namespace battleutils
             if (giveTPtoVictim)
             {
                 //account for attacker's subtle blow which reduces the baseTP gain for the defender
-                float sBlowMult = ((100.0f - dsp_cap((float)PAttacker->getMod(Mod::SUBTLE_BLOW), 0.0f, 50.0f)) / 100.0f);
+                float sBlowMult = ((100.0f - (dsp_cap((float)PAttacker->getMod(Mod::SUBTLE_BLOW), 0.0f, 50.0f) + dsp_cap((float)PAttacker->getMod(Mod::SUBTLE_BLOW_II), 0.0f, 25.0f))) / 100.0f);
```
Comment (Hozu) [inline src/map/utils/battleutils.cpp]: Is Subtle Blow II supposed to cap at +25? As in, you can't do something like, 40 Subtle Blow + 35 Subtle Blow II to hit the limit of 75? Same deal later on in this file, line 2141.

Comment (Fizztastic) [inline src/map/utils/battleutils.cpp]: Do you think it's a good assumption to think that it would work the same as PDT/MDT where MDT+MDT II caps at 87.5%?

Comment (Fizztastic) [inline src/map/utils/battleutils.cpp]: Regarding Cure Potency II, the info I found seems to suggest that there's no cap so I'll be removing that check. 
`Equipment that has "Cure potency II" as a stat will increase beyond the 50% cap. `

There is also another known cap that's not currently being enforced but adding that to all the healing spells shouldn't be too much work.
`The casted and received potency caps are separate and multiply with each other for a maximum possible potency multiplier of 1.95 (1.5 x 1.3).`

**Comment by teschnei** [comment]
Cure potency is enforced (of course), it's just the cure received that's
not capped right now (it caps at 30%), but they do not have a combined cap
if that's what you were insinuating

**Comment by Fizztastic** [comment]
I was only insinuating about the Cure Potency received.

File: src/map/utils/battleutils.cpp [hunk 2]
```diff
@@ -4236,6 +4238,8 @@ namespace battleutils
         resist = 1.0f + ( floor( 256.0f * ( PDefender->getMod(Mod::DMGPHYS) / 100.0f ) ) / 256.0f )
                       + ( floor( 256.0f * ( PDefender->getMod(Mod::DMG)     / 100.0f ) ) / 256.0f );
         resist = dsp_cap(resist, 0.5f, 1.5f); //assuming if its floored at .5f its capped at 1.5f but who's stacking +dmgtaken equip anyway???
+        resist = resist + (floor(256.0f * (PDefender->getMod(Mod::DMGPHYS_II) / 100.0f)) / 256.0f);
```
Comment (teschnei) [inline src/map/utils/battleutils.cpp]: physical damage taken II is additive, not multiplicative with physical damage taken

File: scripts/globals/magic.lua [hunk 3]
```diff
@@ -205,6 +205,8 @@ function getCureFinal(caster,spell,basecure,minCure,isBlueMagic)
         potency = 1.5;
     end
 
+    potency = potency + (caster:getMod(MOD_CURE_POTENCY_II) / 100);
```
Comment (teschnei) [inline scripts/globals/magic.lua]: cure potency II is also additive with cure potency

Comment (Hozu) [inline scripts/globals/magic.lua]: Erm, it's done that way. Cure potency is added and capped at 1.5 above this.

**Comment by teschnei** [comment]
ah, oops

PDT still needs to be additive, though (makes me wonder if MDT is)

**Comment by Hozu** [comment]
Oh damn, I guess I did MDT-% II wrong way back when I added it... damn it. The way Subtle Blow II was done is how it should be.

**Comment by teschnei** [comment]
oh actually, MDT II is multiplicative.. lol

**Pull request closed without merging**